src/clj_fast/inline.clj

```diff
@@ -217,11 +217,37 @@
   For more than one path-value pair this macro will reorder code."
   [m & ksvs]
   {:pre [(every? u/simple-seq? (take-nth 2 ksvs))]}
-  (lens/put-many
+  (lens/update-many
    (fn [m k v] `(c/assoc ~m ~k ~v))
    (fn [m k] `(c/get ~m ~k))
    m ksvs))
 
+(defmacro update-in->
+  "Like update-in but inlines the calls when a static sequence of keys is
+  provided.
+  Can take an unlimited number of [ks exprs] pairs.
+  Unlike update-in, expressions are not & args, i.e.:
+  (update-many-in m [:a :b] f [:a :b] (g 1))
```

Comment:
Are both [:a :b] paths here supposed to be the same? 

Actually, the explode step only allows one action per path, which is probably not what the user expects:
```clj
(update-in-> {:a {:b 1}}
  [:a :b] inc
  [:a :b] str)
;; => {:a {:b "1"}}
```
Maybe the preconditions should include `(distinct? (take-nth 2 ks-exprs))`

Reply:
Nice catch on the docstring typo.
There's also no guarantee of order, so it's a good thing to add this precondition